examples/let-expression/let_expression.bal

```diff
@@ -0,0 +1,47 @@
+import ballerina/io;
+
+public function main() {
+
+    // Following is a simple let expression which involves only a single
+    // variable declaration.
+    int a = let int b = 1 in b * 2;
+    io:println("a: ", a);
+
+    // Declare multiple variables in a let expression.
+    string greeting = let string hello = "Hello ",
+                      string ballerina = "Ballerina!"
+                      in hello + ballerina;
+    io:println("greeting: ", greeting);
+
+    // The scope of a variable declared in a let expression is from
+    // just after the declaration to the end of the let expression.
+    int three = let int one = getInt(), int two = one + one in one + two;
+    io:println("three: ", three);
+
+    // Use let expression with tuple binding pattern.
+    [int, int] v1 = [10, 20];
+    int tupleBindingResult = let [int, int] [d1, d2] = v1,
+                             int d3 = d1 + d2 in  d3 * 2;
+    io:println("tupleBindingResult: ", tupleBindingResult);
+
+    // Use let expression with record binding pattern.
```

Comment:
Not sure if we need another comment if we change the above to remove tuple. 

Specifically saying tuple and record and then ignoring error kinda seems incomplete.

Reply:
Remove the record binding pattern comment and added error binding pattern example.